src/altinn-app-frontend/src/features/form/components/Label.tsx

```diff
@@ -30,13 +30,18 @@ export default function Label(props: IFormLabelProps) {
           data-testid={`label-${props.id}`}
         >
           {props.labelText}
-          {props.labelSettings?.optionalIndicator === false ||
-          props.required ||
-          props.readOnly ? null : (
+          {/* Mark required fields (unless field is readOnly) */}
+          {props.required && !props.readOnly ? (
             <span className='label-optional'>
-              {` (${getLanguageFromKey('general.optional', props.language)})`}
+              {` ${getLanguageFromKey('form_filler.required_label', props.language)}`}
             </span>
-          )}
+          ) : null}
+          {/* Mark optional fields only if optionalIndicator===true (and field is not readOnly) */}
+          {props.labelSettings?.optionalIndicator && !props.required && !props.readOnly ? (
```

Comment:
This is just a suggestion really, but I find it helps to make it clearer to read the code if this check is extracted to before the return statement, to something like this:

```jsx
// before return statement:
const shouldShowOptionalIndicator = props.labelSettings?.optionalIndicator && !props.required && !props.readOnly;
```

```suggestion
          {shouldShowOptionalIndicator ? (
```

then I think the comment also gets a bit redundant.

This could also be done for the check on line 34, to something like `shouldShowRequiredLabel`